pinot-core/src/main/java/org/apache/pinot/core/data/order/OrderByUtils.java

```diff
@@ -0,0 +1,286 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.data.order;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.collections.comparators.ComparableComparator;
+import org.apache.pinot.common.request.AggregationInfo;
+import org.apache.pinot.common.request.SelectionSort;
+import org.apache.pinot.common.utils.BytesUtils;
+import org.apache.pinot.common.utils.DataSchema;
+import org.apache.pinot.common.utils.primitive.ByteArray;
+import org.apache.pinot.core.data.table.Record;
+import org.apache.pinot.core.data.table.Table;
+import org.apache.pinot.core.query.aggregation.function.AggregationFunction;
+import org.apache.pinot.core.query.aggregation.function.AggregationFunctionUtils;
+
+import static org.apache.pinot.common.utils.DataSchema.*;
+
+
+/**
+ * Helper methods to perform order by of list of {@link Record}
+ */
+public class OrderByUtils {
+
+  /**
+   * Constructs a comparator for ordering by the keys in the TableRecord::keys
+   */
+  public static Comparator<Record> getKeysComparator(DataSchema dataSchema,
```

Comment:
Few questions:
1. Why is this a utility as opposed to a class that can be instantiated?
2. It seems that the concept of keys and values has leaked here. Does the Ordering class need to distinguish between the two (the code seems to be quite similar between the two)?

Reply:
OrderByUtils simply provides comparators, given order by and aggregations. It maintains no state of its own. So I thought it suited more as Utils.
I imagine we would have an  OrderByAndTrimmingExecutor which would also make use of these utils

Since OrderByUtils provides comparators for `Record`, I think it is okay that it knows about keys and values, because the comparators built are very different depending on selection/group by/distinct. We could rename them to getSelectionComparator, getGroupByComparator and getDistinctComparator, but I felt the keys/values names are more generic..